checker.cpp

```diff
@@ -1,22 +1,15 @@
-#include <assert.h>
 #include <iostream>
-using namespace std;
-
-bool batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    cout << "Temperature out of range!\n";
-    return false;
-  } else if(soc < 20 || soc > 80) {
-    cout << "State of Charge out of range!\n";
-    return false;
-  } else if(chargeRate > 0.8) {
-    cout << "Charge Rate out of range!\n";
-    return false;
-  }
-  return true;
-}
-
-int main() {
-  assert(batteryIsOk(25, 70, 0.7) == true);
-  assert(batteryIsOk(50, 85, 0) == false);
+#include "BatteryConditions.h"
+#include <assert.h>
+int main()
+{
+BatteryConditions BatteryConditionCheck;
+   assert(BatteryConditionCheck.batteryIsOk(25, 70, 0.7) == true);
```

Comment:
Test cases need to be strengthened to cover all boundary values and in range and out of range values.